pages/search/search.jsx

```diff
@@ -51,24 +50,97 @@ export default React.createClass({
       location[`query`] = { keyword: query };
     }
 
+    // note browserHistory.push() triggers component re-render
     browserHistory.push(location);
+  },
+  clearSearch() {
+    this.refs.searchInput.setState({value: ``}, () => {
+      this.updateBrowserHistory();
+    });
+  },
+  fetchData(params = {}, entries = [], apiPageIndex = 1) {
+    if (!params.search) {
+      // if no search keyword was passed reset search page back to its initial state
+      this.setState(this.getInitialState());
+      return;
+    }
+
+    params.page = apiPageIndex;
+
+    Service.entries
+      .get(params)
+      .then((data) => {
+        entries = entries.concat(data.results);
 
+        if (data.next) {
+          // there are more matched entries in the database we need to fetch
+          this.fetchData(params, entries, apiPageIndex+1);
```

Comment:
@mmmavis 

1. This is standard API behavior and the docs at the top of [the entries endpoint](https://network-pulse-api-staging.herokuapp.com/entries/) already document the `page_size`, so it doesn't make sense to change this 

    `?page_size=` - Number of results on a page. Defaults to 48

2. Yeah I guess I question the purpose of the "Load more" button at all. Only thing I can think of is to avoid forcing the browser to load all those images unless they're wanted. Maybe we do fetch them all at once, and the "load more" button is a gate-keeper which then opens up to show the rest of the results at once? Presumably